app/v2/notifications/post_notifications.py

```diff
@@ -47,7 +47,7 @@ def post_notification(notification_type):
                                         notification_type=notification_type,
                                         api_key_id=api_user.id,
                                         key_type=api_user.key_type,
-                                        reference=form.get('reference', None),
+                                        client_reference=form.get('reference', None),
```

Comment:
Actually I don't think so. The persist only ever set the client-reference. Even if that was from the "reference" post body field.

So the data would not be overwritten.